### ISSUE TITLE
Function-Request: Weighted-Split by Date

### ISSUE THREAD

**Issue by MaNyh** [eneiluj/cospend-nc#27]
See: https://gitlab.com/eneiluj/cospend-nc/-/issues/11

I am not using Cospend because of this. I usually calculate a weighted balance for each month and then I am calculating necessary transactions to sort things out. As I am using my own code, no other member of our shared-household is able to use the programm. 

This function would make Cospend usable for this or similiar kinds of problems.

**Comment by whtv** [comment]
Yes please! To be honest, I don’t even quite understand the usefulness of weights as they are right now. If you change them, all past settlements change as well. That doesn’t make any sense to me. I can see how weight per date is kind of complicated, but how about this: Make weight a per-settlement option, rather than a static user property; or maybe allow the pre-set user weight to be overridden at settlement time?

Also, seeing as there is an option for bills to ignore weights, I think settlement bills should always, or at least by default, do that.

**Comment by whtv** [comment]
After reading #14, I can see how a ‘settle from‘ date is problematic as it kind of contradicts the point of settlement. What if every bill had a ‘settled’ flag instead? The default would then be to only settle unsettled bills. An option at settle time to also include already settled bills (of course taking into account previous settlement bills), in case you change your mind about the weights afterwards, should then result in the current behaviour.

**Comment by Ezwen** [comment]
Hi there, just wanted to share my own use case, a bit different from the scenario above. In my household, we split the expenses based on the income of each member (ie. someone with a higher salary will contribute more). Our current problem with Cospend is that the income of any given member can change during the year. This happened several times already, and each time we have to recreate a new "Project"  to avoid wreaking havoc when changing weights. 

As a workaround, we can create a new project each time there is a chance in income. While creating a new project is simple, the problem is that we each member need to setup one again all the recurrent expenses (internet provider, etc.), with the risk of forgetting one. This is why we would be extremely interested in a way to change the weights without affecting the past expenses.

**Comment by schuandr** [comment]
> Hi there, just wanted to share my own use case, a bit different from the scenario above. In my household, we split the expenses based on the income of each member (ie. someone with a higher salary will contribute more). Our current problem with Cospend is that the income of any given member can change during the year, and that changing weights in the project wreaks havoc in all past expenses.
> 
> As a workaround, we can create a new project each time there is a chance in income. While creating a new project is simple, the problem is that we each member need to setup one again all the recurrent expenses (internet provider, etc.), with the risk of forgetting one. This is why we would be extremely interested in a way to change the weights without affecting the past expenses.

I also have exactly this same use case and would very much appreciate a way to do this. Or has anyone perhaps figured out a workaround that doesn't involve setting up a new project every time the weighting changes?
